src/udiskslinuxfilesystem.c

```diff
@@ -1983,10 +1983,497 @@ handle_set_label (UDisksFilesystem      *filesystem,
 
 /* ---------------------------------------------------------------------------------------------------- */
 
+/* runs in thread dedicated to handling method call */
+static gboolean
+handle_resize (UDisksFilesystem      *filesystem,
+               GDBusMethodInvocation *invocation,
+               guint64                size,
+               GVariant              *options)
+{
+  UDisksBlock *block;
+  UDisksObject *object = NULL;
+  UDisksDaemon *daemon = NULL;
+  const gchar *probed_fs_usage;
+  const gchar *probed_fs_type;
+  BDFsResizeFlags mode;
+  const gchar *action_id;
+  const gchar *message;
+  uid_t caller_uid;
+  gid_t caller_gid;
+  GError *error = NULL;
+  UDisksBaseJob *job = NULL;
+  gchar *required_utility = NULL;
+  const gchar * const *existing_mount_points;
+
+  g_mutex_lock (&UDISKS_LINUX_FILESYSTEM (filesystem)->lock);
```

Comment:
Maybe this can acquire the lock a bit later? I don't think it's necessary to hold it while asking user for password, etc.